python/snewpy/models/ccsn.py

```diff
@@ -308,17 +297,19 @@ class Walk_2019(_RegistryModel):
     from the `Garching Supernova Archive`_.
     """
 
-    param = {'progenitor_mass': 40 * u.Msun}
-    param_combinations = get_param_combinations(param)
-
+    param = {'progenitor_mass': [40., 75.] * u.Msun,
+             'direction': [1,2,3]}
+    _param_validator = lambda p: (p['progenitor_mass'] == 75. * u.Msun and p['direction'] in (1,2)) or \
+        (p['progenitor_mass'] == 40. * u.Msun and p['direction'] in (1,2,3))
+    
     @_warn_deprecated_filename_argument
     def __new__(cls, filename=None, eos='LS220', *, progenitor_mass=None):
         if filename is not None:
             # Metadata creation is implemented in snewpy.models.base._GarchingArchiveModel
             return loaders.Walk_2019(os.path.abspath(filename))
 
-        check_valid_params(cls, progenitor_mass=progenitor_mass)
-        filename = f's{progenitor_mass.value:3.1f}c_3DBH_dir1'
+        cls.check_valid_params(cls, progenitor_mass=progenitor_mass)
```

Comment:
Similar to `Walk_2018`, the check should include the new parameter
```suggestion
        cls.check_valid_params(cls, progenitor_mass=progenitor_mass, direction=direction)
```